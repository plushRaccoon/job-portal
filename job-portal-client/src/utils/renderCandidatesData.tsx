// import React, { useState } from 'react';
// import { List, ListItem, IconButton, ListItemText, Box, ListItemIcon, TextField, Button, Modal } from "@mui/material";
// import Divider from "@mui/material/Divider";
// import DeleteIcon from '@mui/icons-material/Delete';
// import BorderColorRoundedIcon from '@mui/icons-material/BorderColorRounded';
// import { StyleForModals } from './constants';
// import EditApplicationForm from '../components/application-form';

// export const RenderCandidatesData = ({ item }) => (
//   <>
//     {item.firstName} {item.lastName}
//     <Divider orientation="vertical" flexItem style={{ margin: '0 8px' }} />
//     {item.email}
//     <Divider orientation="vertical" flexItem style={{ margin: '0 8px' }} />
//     {item.phone}
//   </>
// )

// export const RenderCandidatesDetails = ({ item, onDelete, onEdit }) => {
//   // const [editedCV, setEditedCV] = useState('');
//   // const [openModal, setOpenModal] = useState(false);
//   // const [ applicationId, setApplicationId ] = useState('');
//   const [openEditModal, setOpenEditModal] = useState(false);
//   const [editApplication, setEditApplication] = useState(null);

//   // const handleEditClick = (id, cv) => {
//   //   setOpenModal(true);
//   //   setApplicationId(id);
//   //   setEditedCV(cv);
//   // };

//   // const handleCancelEdit = () => {
//   //   setOpenModal(false);
//   //   setEditedCV('');
//   // };

//   // const handleSubmitEdit = (editedCV) => {
//   //   onEdit('applications', applicationId, editedCV);
//   //   setOpenModal(false);
//   //   setEditedCV('');
//   // };

//   const handleEditClick = (application) => {
//     setEditApplication(application);
//     setOpenEditModal(true);
//   };

//   const handleCloseEditModal = () => {
//     setOpenEditModal(false);
//     setEditApplication(null);
//   };

//   const handleSubmitEdit = (id, editedData) => {
//     onEdit('applications', id, editedData);
//     setOpenEditModal(false);
//     setEditApplication(null);
//   };

//   return (
//     <>
//       {item.applications?.map((application) => {
//         const { id, cv, position: { title, status } } = application;
//         return (
//           <List key={id}>
//             <ListItem secondaryAction={
//               <IconButton edge="end" aria-label="delete" color='error' onClick={() => onDelete('applications', id)}>
//                 <DeleteIcon />
//               </IconButton>
//             }>
//               <ListItemText sx={{ marginRight: 10 }}>
//                 <Box key={id}>
//                   <Box display="flex" alignItems="center" marginBottom="10px">
//                     {title}
//                     <Divider orientation="vertical" flexItem style={{ margin: '0 8px' }} />
//                     {status}
//                   </Box>
//                   {cv}
//                 </Box>
//               </ListItemText>
//               <ListItemIcon>
//                   {/* <IconButton size="large" aria-label="edit" color="default" onClick={() => handleEditClick(id, cv)}> */}
//                   <IconButton size="large" aria-label="edit" color="default" onClick={() => handleEditClick(application)}>
//                     <BorderColorRoundedIcon />
//                   </IconButton>
//               </ListItemIcon>
//             </ListItem>
//           </List>
//         );
//       })}
//       {editApplication && (
//         <EditApplicationForm
//           open={openEditModal}
//           handleClose={handleCloseEditModal}
//           application={editApplication}
//           handleSubmit={handleSubmitEdit}
//         />
//       )}
//       {/* <Modal
//         open={openModal}
//         onClose={handleCancelEdit}
//         aria-labelledby="edit-cv-form-title"
//         aria-describedby="edit-cv-form-description"
//       >
//         <Box sx={StyleForModals}>
//           <TextField
//             margin="normal"
//             required
//             fullWidth
//             multiline
//             id="cv"
//             label="CV"
//             name="cv"
//             value={editedCV}
//             onChange={(e) => setEditedCV(e.target.value)}
//           />
//           <Button onClick={() => handleSubmitEdit(editedCV)} variant="contained" color="primary" fullWidth size='large'>
//             Submit
//           </Button>
//         </Box>
//       </Modal> */}
//     </>
//   );
// };


import { useState } from 'react';
import { Box, List, ListItem, ListItemText, ListItemIcon, IconButton, Divider } from '@mui/material';
import DeleteIcon from '@mui/icons-material/Delete';
import BorderColorRoundedIcon from '@mui/icons-material/BorderColorRounded';
import EditApplicationForm from '../components/application-form'; // Ensure the import is correct

export const RenderCandidatesData = ({ item }) => (
    <>
      {item.firstName} {item.lastName}
      <Divider orientation="vertical" flexItem style={{ margin: '0 8px' }} />
      {item.email}
      <Divider orientation="vertical" flexItem style={{ margin: '0 8px' }} />
      {item.phone}
    </>
  )

export const RenderCandidatesDetails = ({ item, onDelete, onEdit }) => {
  const [openEditModal, setOpenEditModal] = useState(false);
  const [editApplication, setEditApplication] = useState(null);

  const handleEditClick = (application) => {
    setEditApplication(application);
    setOpenEditModal(true);
  };

  const handleCloseEditModal = () => {
    setOpenEditModal(false);
    setEditApplication(null);
  };

  const handleSubmitEdit = (id) => {
    onEdit('applications', id, editApplication);
    setOpenEditModal(false);
    setEditApplication(null);
  };

  return (
    <>
      {item.applications?.map((application) => {
        const { id, cv, position: { title, status } } = application;
        return (
          <List key={id}>
            <ListItem
              secondaryAction={
                <IconButton edge="end" aria-label="delete" color="error" onClick={() => onDelete('applications', id)}>
                  <DeleteIcon />
                </IconButton>
              }
            >
              <ListItemText sx={{ marginRight: 10 }}>
                <Box key={id}>
                  <Box display="flex" alignItems="center" marginBottom="10px">
                    {title}
                    <Divider orientation="vertical" flexItem style={{ margin: '0 8px' }} />
                    {status}
                  </Box>
                  {cv}
                </Box>
              </ListItemText>
              <ListItemIcon>
                <IconButton size="large" aria-label="edit" color="default" onClick={() => handleEditClick(application)}>
                  <BorderColorRoundedIcon />
                </IconButton>
              </ListItemIcon>
            </ListItem>
          </List>
        );
      })}
      {editApplication && (
        <EditApplicationForm
          open={openEditModal}
          handleClose={handleCloseEditModal}
          application={editApplication}
          handleSubmit={() => handleSubmitEdit(editApplication.id)}
        />
      )}
    </>
  );
};
