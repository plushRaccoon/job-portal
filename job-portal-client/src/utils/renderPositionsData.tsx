import React from "react";
import { List, ListItem, ListItemText, Box } from "@mui/material";
import Divider from "@mui/material/Divider";

export const RenderPositionsData = ({ item }) => {
  return (
  <>
    {item.title}
    <Divider orientation="vertical" flexItem style={{ margin: '0 8px' }} />
    {item.status}
    <Divider orientation="vertical" flexItem style={{ margin: '0 8px' }} />
    Applied: {item.applications?.length || 0}
  </>
)}

export const RenderPositionsDetails = ({item}) => {
  return (
  <>
    {item.applications?.map((application) => {
      const { candidate: { firstName, lastName, email, phone }, cv, id } = application;
      return (
        <List key={id}>
          <ListItem>
            <ListItemText>
                <Box key={id}>
                  <Box display="flex" alignItems="center" marginBottom="10px">
                    {firstName} {lastName}
                    <Divider orientation="vertical" flexItem style={{ margin: '0 8px' }} />
                    {email}
                    <Divider orientation="vertical" flexItem style={{ margin: '0 8px' }} />
                    {phone}
                  </Box>
                  {cv}
                </Box>
            </ListItemText>
          </ListItem>
        </List>
      );
    })}
  </>
)}
