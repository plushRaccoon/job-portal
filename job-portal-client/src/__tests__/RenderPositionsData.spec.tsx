import React from 'react';
import { render } from '@testing-library/react';
import { RenderPositionsData, RenderPositionsDetails } from '../utils/renderPositionsData';
import '@testing-library/jest-dom';

const mockPosition = {
  title: 'Software Engineer',
  status: 'Active',
  applications: [
    {
      id: '1',
      candidate: {
        firstName: 'John',
        lastName: 'Doe',
        email: 'john.doe@example.com',
        phone: '123-456-7890',
      },
      cv: 'John Doe CV',
    },
  ],
};

describe('RenderPositionsData Component', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<RenderPositionsData item={mockPosition} />);
    
    expect(asFragment()).toMatchSnapshot();
  });
});

describe('RenderPositionsDetails Component', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<RenderPositionsDetails item={mockPosition} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
