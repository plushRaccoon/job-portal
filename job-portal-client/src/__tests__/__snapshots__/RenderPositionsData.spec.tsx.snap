// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`RenderPositionsData Component > matches snapshot 1`] = `
<DocumentFragment>
  Software Engineer
  <hr
    class="MuiDivider-root MuiDivider-fullWidth MuiDivider-vertical MuiDivider-flexItem css-1tk1bks-MuiDivider-root"
    style="margin: 0px 8px;"
  />
  Active
  <hr
    class="MuiDivider-root MuiDivider-fullWidth MuiDivider-vertical MuiDivider-flexItem css-1tk1bks-MuiDivider-root"
    style="margin: 0px 8px;"
  />
  Applied: 1
</DocumentFragment>
`;

exports[`RenderPositionsDetails Component > matches snapshot 1`] = `
<DocumentFragment>
  <ul
    class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
  >
    <li
      class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
    >
      <div
        class="MuiListItemText-root css-tlelie-MuiListItemText-root"
      >
        <span
          class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
        >
          <div
            class="MuiBox-root css-0"
          >
            <div
              class="MuiBox-root css-1rwzck"
            >
              John Doe
              <hr
                class="MuiDivider-root MuiDivider-fullWidth MuiDivider-vertical MuiDivider-flexItem css-1tk1bks-MuiDivider-root"
                style="margin: 0px 8px;"
              />
              john.doe@example.com
              <hr
                class="MuiDivider-root MuiDivider-fullWidth MuiDivider-vertical MuiDivider-flexItem css-1tk1bks-MuiDivider-root"
                style="margin: 0px 8px;"
              />
              123-456-7890
            </div>
            John Doe CV
          </div>
        </span>
      </div>
    </li>
  </ul>
</DocumentFragment>
`;

exports[`RenderPositionsDetails Component > renders applications correctly 1`] = `
<DocumentFragment>
  <ul
    class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
  >
    <li
      class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
    >
      <div
        class="MuiListItemText-root css-tlelie-MuiListItemText-root"
      >
        <span
          class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
        >
          <div
            class="MuiBox-root css-0"
          >
            <div
              class="MuiBox-root css-1rwzck"
            >
              John Doe
              <hr
                class="MuiDivider-root MuiDivider-fullWidth MuiDivider-vertical MuiDivider-flexItem css-1tk1bks-MuiDivider-root"
                style="margin: 0px 8px;"
              />
              john.doe@example.com
              <hr
                class="MuiDivider-root MuiDivider-fullWidth MuiDivider-vertical MuiDivider-flexItem css-1tk1bks-MuiDivider-root"
                style="margin: 0px 8px;"
              />
              123-456-7890
            </div>
            John Doe CV
          </div>
        </span>
      </div>
    </li>
  </ul>
</DocumentFragment>
`;
